### PR TITLE
New blankslate hoc with error

### PR DESCRIPTION
**Pull request by pochretien**
### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

image

### CODE REVIEW

File: packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import * as VaporSVG from 'coveo-styleguide';
+import * as React from 'react';
+import {Svg} from '../svg/Svg';
+import {IBlankSlateProps} from './BlankSlate';
+
+export const blankSlateWithError = () => (
```
Comment (gdostie) [inline packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx]:
```suggestion
export const blankSlateWithError = (
```
That way, you wouldn't have to call the function
```ts
export const BlankSlateWithTableInError = _.compose(
    blankSlateWithTable(),
    blankSlateWithError,
)(BlankSlate);
```

File: packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Component: React.ComponentClass<IBlankSlateProps>
+): React.ComponentClass<IBlankSlateProps> => {
+    class ComponentWithTable extends React.PureComponent<IBlankSlateProps> {
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx]:
```suggestion
    class BlankSlateWithError extends React.PureComponent<IBlankSlateProps> {
```

File: packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                    description={<span className="text-black semibold">{this.props.description}</span>}
+                    classes={['border-color-orange-8 mod-error']}
+                >
+                    {this.props.children}
+                </Component>
+            );
+        }
+    }
+
+    return ComponentWithTable;
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/blankSlate/BlankSlateWithError.tsx]:
```suggestion
    return BlankSlateWithError;
```

File: packages/react-vapor/docs/src/components/ComponentPage.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ const ComponentPage: React.FunctionComponent<ComponentPageProps> = (props) => {
         {id: getTabId('development'), title: component.firstTabLabel || 'Develop'},
         ...tabs.sort((tabA: TabConfig, tabB: TabConfig) => tabA.order - tabB.order).map(mapTabConfigToProps),
     ];
-    const PageLayout = hasMarkdownTabs ? PageLayoutWithTabs : PageLayoutWithoutTabs;
+    const PageLayout: any = hasMarkdownTabs ? PageLayoutWithTabs : PageLayoutWithoutTabs;
```
Comment (gdostie) [inline packages/react-vapor/docs/src/components/ComponentPage.tsx]: If you don't wanna type this to any, you could modify the line 13:
```tsx
type ComponentPageProps = IComponent & {tabs?: TabConfig[]};
```

**Pull request merged**